app/src/main/java/tech/ula/utils/DownloadUtility.kt

```diff
@@ -57,18 +64,53 @@ class DownloadUtility(val context: Context, val archType: String, distType: Stri
         return downloadManager.enqueue(request)
     }
 
-    private fun assetNeedsToUpdated(type: String): Boolean {
+    private fun assetNeedsToUpdated(type: String, endpoint: String, newFilesystem: Boolean): Boolean {
+        val sharedPref = context.getSharedPreferences("file_date_stamps", Context.MODE_PRIVATE)
+
+        //TODO: make it so we download a full group of files, if any has changed (not the rootfs though, that is special)
+        //This will take care of a few possible corner cases
+
+        //only download the rootfs the first time, it cannot simply be updated like other files
+        if(type.contains("rootfs") && !newFilesystem) {
+            return false
+        }
+
+        //only download if this is a newFilesystem or it has been more than a day since we last checked
+        val now = java.util.Date().time
+        //TODO: should be store last updated in the session itself
+        val lastUpdated = sharedPref.getLong("lastUpdated", 0)
+        if (now < (lastUpdated + 86400000L) && !newFilesystem) {
+            return false
+        } else {
+            with (sharedPref.edit()) {
+                putLong("lastUpdated", now)
+                commit()
+            }
+        }
+
         val (subdirectory, filename) = type.split(":")
         val asset = File("${context.filesDir.path}/$subdirectory/$filename")
-        // TODO more sophisticated version checking
+
+        //update a file if a connection can be made and if the datestamp is different
+        val currentDateStamp = sharedPref.getLong(type, 0)
+        val newDateStamp = urlDateModified(endpoint)
+        if ((newDateStamp != 0L) && (currentDateStamp != newDateStamp)) {
```

Comment:
What does `newDateStamp != 0L` check for?

Reply:
If it returns 0L it means it failed to make connection with the server when checking the timestamp (or the path is busted).  in those scenarios we should just stick with the version we already have.